### ISSUE TITLE
feature request: able to change taskbar text color

### ISSUE THREAD

**Issue by mailinglists35** [ethanhs/TranslucentTB#41]
I don't know if TranslucentTB can do this, but I'm just asking, feel free to close if it's out of scope!
I am running a less common configuration: vertical taskbar on right, autohide, small text, never combine.

The problem is that when the background is lighter, windows keeps displaying taskar text items with white color. This makes it from difficult to impossible - depending on what background color or window is below the taskbar - to read the text on taskbar with TTB on.

Attaching screenshots. The question is: is there any way TTB can outsmart windows and make taskbar text darker when the background is lighter? I've tried all possible combinations in windows settings and it seems it insists on keeping the white text no matter what (except when using high contrast which is seriously ugly and makes transparency pointless)

![image](https://cloud.githubusercontent.com/assets/2054302/22502761/75ac150a-e87f-11e6-9264-9ec5ed872d5b.png)
![image](https://cloud.githubusercontent.com/assets/2054302/22502778/807b348e-e87f-11e6-8084-f0019fa9cb68.png)

**Comment by MrAksel** [comment]
I don't know if its possible to change the text color easily, but you do have some other workarounds possible. The simplest solution would be to make the taskbar a little darker, and TB supports that. If you make a shortcut to the executable, you can edit what arguments/options are passed to the program at launch. You right click the shortcut, press "Properties", and in the "Target" field you can append arguments. To change the color you use the --tint [color] option, where [color] is in hex format. You could for example append "--tint 60606060" to have a partly transparent gray taskbar. If anything is unclear just ask, I hope this helps! 

ps.: The next version will include a configuration file that makes it easier to adjust these options. There are other options too documented in the usage.md file.

**Comment by charlesmilette** [comment]
I know that Classic Shell can do this, but then there's no point having
TranslucentTB.

Le mer. 1 févr. 2017 05:14, mailinglists35 <notifications@github.com> a
écrit :

**Comment by ethanhs** [comment]
Yeah, classic shell can do this. The other thing I can think of is changing the theme for Windows, which has some control of the taskbar style, however that opens up its own can of worms. I'd like to do this, but I'm not sure how I can accomplish this.

**Comment by charlesmilette** [comment]
An old version of the Classic Shell source code is available. Might be worth investigating.

De : Ethan
Envoyé le :1 février 2017 12:17
À : ethanhs/TranslucentTB
Cc : Charles Milette; Comment
Objet :Re: [ethanhs/TranslucentTB] feature request: able to change taskbartext color (#41)

Yeah, classic shell can do this. The other thing I can think of is changing the theme for Windows, which has some control of the taskbar style, however that opens up its own can of worms. I'd like to do this, but I'm not sure how I can accomplish this.

**Comment by ethanhs** [comment]
Yes, I have it downloaded, but I think the feature was added in a later version, sadly. I think the oss ended at 3.6.8 and the feature was added in around 4.2 or so.

**Comment by ethanhs** [comment]
I need to test using WM_SETFONT and dealing with the DC for the taskbar, I'm guessing that is how ClassicShell is doing it.

**Comment by ethanhs** [comment]
Im looking into this. I think I can figure out how it works.

**Comment by gerbil124** [comment]
I tried to add --tint 60606060 parameter to have a partly transparent gray taskbar, but i have 2 screens and it works on just one the other one is full transparent(full clear). The parameter made it awesome but just on one screen- :(

**Comment by ethanhs** [comment]
Made high priority because a lot of people want this, and it ruins the effect with light wallpapers.

**Comment by sylveon** [comment]
Out of a regular taskbar, the only things that changing text colors will change is this:
![](https://charles.getsharex.com/AngelicTomatoPlanthopper)

Of course on the taskbar mode that displays window names it might be more interesting

**Comment by ethanhs** [comment]
This depends on #134, so sadly I don't think we will be able to do this in V4.

**Comment by Margen67** [comment]
What about a text *shadow*?

**Comment by CharleyMorgan** [comment]
Win10 19H1 will be published before long.The taskbar and menu color will be modified shortly.

**Comment by denosawr** [comment]
I could be just completely clueless, but could you please explain what you mean @CharleyMorgan? 19H1 doesn't seem to have any taskbar modifications scheduled.

**Comment by CharleyMorgan** [comment]
because I found a news(https://www.nruan.com/63557.html).You can follow it or other news lately.

**Comment by denosawr** [comment]
Your news article doesn't mention anything about changing the colour of the text on the taskbar - only the taskbar itself.

**Comment by CharleyMorgan** [comment]
Ohhhhh no,you are right

**Comment by mailinglists35** [comment]
btw I've got a native transparent taskbar in one w10 vm freely downloadable from ie testdrive site. It must have been a bug as it permanently reverted to normal after entering taskbar settings.

**Comment by henrique-eba** [comment]
I would love this feature as well.
I use random wallpapers settings, so would be very nice if the text adapted dynamically too.

The perfect solution (for me) is having the ability to select two colors (for contrast with dark and light colors) and the task bar text color be adjusted dynamically based on the average color from the bottom of the current wallpaper.

Keep rocking! :)

**Comment by cjwijtmans** [comment]
I wonder if inverting the win10 accent colour would be a good alternative.

**Comment by sylveon** [comment]
This is something I've considered but I don't think will be implemented, at least not for Windows 10. There are several issues:

- the only "text" in the taskbar is the clock, any toolbars, and window titles if you disabled collapsing icons. So for most users, this will only change the clock and not achieve the desired result.
- To change the color of other icons (like tray icons and app icons), I have to convince the system that a different color theme is set, while also not making everything else (start menu, action center, volume flyouts etc) think that.
- If I want to also fix the tray icons of third party apps, I also have to fix them in a case by case basis, and risk being broken every time said app updates.

Now, this might be easier to do in Windows 11, since the taskbar is XAML - but I need to figure out how to alter the taskbar's inner XAML content. And it still wouldn't affect third party apps.

**Comment by onisimus** [comment]
I figured it out! You need to Right-click your desktop and go to Personalize > Click Colors on Left panel > then Choose Dark on the drop down color. This swapped my taskbar text from black to white.

**Comment by hungbhv** [comment]
So I inadvertently found that I can change the color of taskbar texts and icons when configuring another app, Auto Dark Mode which is used to schedule dark and light mode. Just fiddling around with these settings let me change the color of texts and icons without affecting the colors of anything else on the system. Have anybody tried this yet? The setting stays when toggling between dark and light mode.

![image](https://github.com/TranslucentTB/TranslucentTB/assets/78145800/206dc871-bdd8-475f-acbe-54018456ce09)

**Comment by MMMMMoris** [comment]
Any update now? We are from 2024😭 This request is still under development😭😭

**Comment by Anvnamnet** [comment]
And now we are in 2026.
I think we can close this now. It's not gonna happen 9 years later!
hahaha

**Comment by 4-the-mankind** [comment]
This is the only thing preventing me to use this app. I have windows dark and a light background

**Comment by Anvnamnet** [comment]
> This is the only thing preventing me to use this app. I have windows dark and a light background

I don't know if you seen how long this has been a thing and never fixed. It was reported 2017 (9 years ago), so I dont believe anything will happen.
